chingsley/ireporter-react
# #165168074 Setup the react-redux workflow

#### What does this PR do?
Sets up a react-redux workflow.

#### Description of Task to be completed?
- adds actions and reducer directories with a dummy reducer and basic signup/sign in actions
- adds redux-thunk for asynchronous actions
- adds react-redux and wrap the AppRouter in the Provider HOC.
- configure the store with the combined reducers.
- Create the page templates with simple descriptive headings

#### How should this be manually tested?
- run `npm run start:dev`, open the browser and open redux devtools extension. You should be able to see the reducers I added.
- Click on the links to navigate different page templates.

#### Any background context you want to provide?
N/A

### Any relevant pivotal tracker stories
https://www.pivotaltracker.com/story/show/165168074
